src/include/stir/IO/InputFileFormatSAFIR.h

```diff
@@ -0,0 +1,147 @@
+/* InputFileFormatSAFIR.h
+
+ Class defining input file format for coincidence listmode data for SAFIR. 
+ Jannis Fischer
+ jannis.fischer@cern.ch
+
+	Copyright 2015 ETH Zurich, Institute of Particle Physics
+
+	Licensed under the Apache License, Version 2.0 (the "License");
+	you may not use this file except in compliance with the License.
+	You may obtain a copy of the License at
+
+		http://www.apache.org/licenses/LICENSE-2.0
+
+	Unless required by applicable law or agreed to in writing, software
+	distributed under the License is distributed on an "AS IS" BASIS,
+	WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+	See the License for the specific language governing permissions and
+	limitations under the License.
+ */
+
+/*!
+
+  \file
+  \ingroup listmode
+  \brief Declaration of class stir::SAFIRCListmodeInputFileFormat
+
+  \author Jannis Fischer
+*/
+
+#ifndef __local_SAFIR_InputFileFormatSAFIR_H__
+#define __local_SAFIR_InputFileFormatSAFIR_H__
+
+#include <cstring>
+#include <string>
+#include <iostream>
+
+#include "stir/IO/InputFileFormat.h"
+#include "stir/utilities.h"
+#include "stir/ParsingObject.h"
+
+#include "stir/listmode/CListRecordSAFIR.h"
+#include "stir/listmode/CListModeDataSAFIR.h"
+
+START_NAMESPACE_STIR
+
+//! Class for reading SAFIR coincidence listmode data.
```

Comment:
best to add doxygen with explicit header as example. also refer to CListModeDataSAFIR here. Also refer to other lasses/places with documentation on file formats for the files here.